tests/editing/totals_and_components/test_totals_and_components.py

```diff
@@ -5686,59 +6031,26 @@ class TestValidateInput:
                 300.0,
                 20,
                 "0.1",
-                6,
                 (
-                    100,
+                    Decimal("100"),
                     [
-                        ComponentPair(original_value=1, final_value=None),
-                        ComponentPair(original_value=2, final_value=None),
-                        ComponentPair(original_value=3, final_value=None),
-                        ComponentPair(original_value=4, final_value=None),
+                        ComponentPair(original_value=Decimal("1"), final_value=None),
+                        ComponentPair(original_value=Decimal("2"), final_value=None),
+                        ComponentPair(original_value=Decimal("3"), final_value=None),
+                        ComponentPair(original_value=Decimal("4"), final_value=None),
                     ],
-                    100.0,
-                    300.0,
-                    20,
-                    0.1,
-                    6,
+                    True,
+                    Decimal("100"),
+                    Decimal("300"),
+                    Decimal("20"),
+                    Decimal("0.1"),
                 ),
                 "Test 19: percentage difference threshold is a string of a number",
                 # Test to ensure percentage difference threshold number entered as a string passes
             ),
-            (
-                "R",
-                100,
-                [
-                    ComponentPair(original_value=1, final_value=None),
-                    ComponentPair(original_value=2, final_value=None),
-                    ComponentPair(original_value=3, final_value=None),
-                    ComponentPair(original_value=4, final_value=None),
-                ],
-                True,
-                100.0,
-                300.0,
-                20,
-                0.1,
-                "6",
-                (
-                    100,
-                    [
-                        ComponentPair(original_value=1, final_value=None),
-                        ComponentPair(original_value=2, final_value=None),
-                        ComponentPair(original_value=3, final_value=None),
-                        ComponentPair(original_value=4, final_value=None),
-                    ],
-                    100.0,
-                    300.0,
-                    20,
-                    0.1,
-                    6,
-                ),
-                "Test 20: precision is a string of a number",
-                # Test to ensure precision number entered as a string passes
-            ),
         ],
     )
-    def test_validate_input(
+    def convert_input_to_decimal(
```

Comment:
Not for this ticket - but the tests for utility functions should be separate from the method.